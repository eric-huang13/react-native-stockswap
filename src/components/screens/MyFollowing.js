import React, {Component} from 'react';
import {
  View,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  SafeAreaView,
  FlatList
} from 'react-native';
import SearchInput from '../../icons/SearchInput';
import {connect} from 'react-redux';
import MyFollowingBox from './MyFollowingBox'


export class MyFollowing extends Component {
  constructor(props) {
    super(props);
    this.state = {
      input: '',
    };
  }

 
  handleChange = (text) => {
    this.setState({input: text});
  };
   accountId = this.props.userAccount.id

   navigationByCondition = item => {
    const {navigation} = this.props;
    if (item.id === this.accountId) {
      navigation.navigate({
        name: 'MyProfile',
        params: {id: item.id},
      })
    } else {
      navigation.navigate({
        name: 'Profile',
        params: {id: item.id}
      })
    }
  };

  render() {
    const {input} = this.state;
    const {users, userAccount} = this.props;

    const accountId = userAccount.id

    //will get actual follower data when backend hooked up
    //just using all people for styling purposes
    const filteredUsers = users.filter((item) =>
      item.name.toLowerCase().includes(input.toLowerCase()),
    );
   
    return (
      <SafeAreaView style={style.mainContainer}>
     
          <View style={style.searchInputContainer}>
            <View
              style={style.searchInputIconContainer}>
              <SearchInput />
            </View>
            <TextInput
              style={style.searchInput}
              placeholder="Search"
              placeholderTextColor="lightgrey"
              onChangeText={(text) => this.handleChange(text)}
            />
          </View>
          <FlatList
            style={style.listContainer}
            data={filteredUsers}
            renderItem={({item, index}) => (
                <TouchableOpacity
                key={item.id}
                onPress={()=>this.navigationByCondition(item)}>
                <MyFollowingBox item={item} index={index}/>
              </TouchableOpacity>
            )}
          />
      </SafeAreaView>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    users: state.people.users,
    userAccount: state.user.userFakeData,
  };
};

export default connect(mapStateToProps)(MyFollowing);

const style = StyleSheet.create({
  mainContainer: { 
    backgroundColor: '#2a334a',
    flex:1,
  },
  searchInputContainer: {
    marginBottom: 15,
   
  },
  searchInput: {
    paddingLeft: 40,
    alignContent: 'center',
    backgroundColor: '#3e4d6c',
    color: 'lightgrey',
    fontSize: 15,
    height: 36,
    fontFamily: 'Montserrat-Italic',
    paddingVertical: 0,
  },
  searchInputIconContainer:{
    position: 'absolute',
    zIndex: 1,
    left: 14,
    top: 10,
  },
listContainer:{

},
});
