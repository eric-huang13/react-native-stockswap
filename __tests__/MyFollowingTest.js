import React from 'react';
import MyFollowing from '../src/components/MyProfileTabScreens/MyFollowing';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6

jest.mock('@react-native-community/google-signin', () => {});

jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<MyFollowing users={[]} userAccount={{img:'test' }}/>);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});