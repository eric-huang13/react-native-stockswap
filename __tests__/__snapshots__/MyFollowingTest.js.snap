// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 21.186131386861312,
      }
    }
  >
    <View
      style={
        Object {
          "left": 19.77372262773723,
          "position": "absolute",
          "top": 14.124087591240876,
          "zIndex": 1,
        }
      }
    >
      <SearchInput />
    </View>
    <Component
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="lightgrey"
      rejectResponderTermination={true}
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#3e4d6c",
          "color": "lightgrey",
          "fontFamily": "Montserrat-Italic",
          "fontSize": 21.186131386861312,
          "height": 50.846715328467155,
          "paddingLeft": 56.496350364963504,
          "paddingVertical": 0,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <FlatList
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</ForwardRef(SafeAreaView)>
`;
